src/ray/object_manager/plasma/store_runner.cc

```diff
@@ -97,13 +97,14 @@ void PlasmaStoreRunner::Start(ray::SpillObjectsCallback spill_objects_callback,
     // large amount of space up front. According to the documentation,
     // dlmalloc might need up to 128*sizeof(size_t) bytes for internal
     // bookkeeping.
-    void *pointer = PlasmaAllocator::Memalign(
-        kBlockSize, PlasmaAllocator::GetFootprintLimit() - 256 * sizeof(size_t));
-    RAY_CHECK(pointer != nullptr);
+    // TODO(scv119): this leaks details of PlasmaAlloctor,
+    // should be part of PlasmaAllocator contruction.
+    auto allocation = PlasmaAllocator::GetInstance().Allocate(
+        PlasmaAllocator::GetInstance().GetFootprintLimit() - 256 * sizeof(size_t));
+    RAY_CHECK(allocation.has_value());
```

Comment:
Can you also add a better error message? 
```suggestion
    RAY_CHECK(allocation.has_value()) << "Plasma store allocation failed because there's not enough /dev/shm available.";
```
or something like that